newlib/libc/tinystdio/posixiob.c

```diff
@@ -40,15 +40,11 @@ static char read_buf[BUFSIZ];
 
 static struct __file_bufio __stdin = FDEV_SETUP_POSIX(0, read_buf, BUFSIZ, __SRD, 0);
 static struct __file_bufio __stdout = FDEV_SETUP_POSIX(1, write_buf, BUFSIZ, __SWR, __BLBF);
+static struct __file_bufio __stderr = FDEV_SETUP_POSIX(2, write_buf, BUFSIZ, __SWR, __BLBF);
```

Comment:
Need a separate buffer for stderr, but it doesn't have to be BUFSIZ bytes, it could be as small as a single byte.